### ISSUE TITLE
darwin-rebuild switch --flake is broken by new nix

### ISSUE THREAD

**Issue by ahmedelgabri** [LnL7/nix-darwin#307]
This is what I get when I try to run `darwin-rebuild switch --flake`

```
$ darwin-rebuild switch --flake .
error: 'info' is not a recognised command
Try 'nix --help' for more information.
```

A quick investigation seems like it fails here on this line https://github.com/LnL7/nix-darwin/blob/67da67bb7a843d48c8cdae6770656c5333e75de0/pkgs/nix-tools/darwin-rebuild.sh#L129

```
$ nix flake info --json
error: 'info' is not a recognised command
Try 'nix --help' for more information.
```

This is an issue on nixpkgs that might be related https://github.com/NixOS/nixpkgs/issues/116938

I think the command changed to `nix flakes metadata` instead. `nix flakes --help` & `nix flakes metadata --help` confirms this.

**Comment by MattPColeman** [comment]
Have you found a workaround for this? Just ran into it on a new laptop, can't install my flake dotfiles 😭

**Comment by hardselius** [comment]
@MattPColeman I have a fix pending merge. You can use my fork in the meantime.

**Comment by MattPColeman** [comment]
> @MattPColeman I have a fix pending merge. You can use my fork in the meantime.

hero

**Comment by Enzime** [comment]
Fixed by #308
